services/src/pro/handlers/users.rs

```diff
@@ -444,6 +445,7 @@ mod tests {
     ) -> ServiceResponse
     where
         C::ProjectDB: ProProjectDb,
+        C::DatasetDB: UpdateDatasetPermissions,
```

Comment:
ideally yes, but the ProContext is just a sub trait of the Context trait. And a Context just requires the Context::DatasetDB to be a DatabaseDb and nothing more.

Reply:
Require a DatasetDb to have `UpdateDatasetPermissions`.

Reply:
Ok, but how?

Reply:
I would say just add the trait to the super trait (Context).

Reply:
the super trait is for non-pro as well, so I can't add it there, because there are no users or permissions in non-pro

Reply:
There is a ProContext trait for that I recall

Reply:
Yes, but the DatasetDb ist defined in the Context trait, which is a super trait of the ProContext

Reply:
Can't we set the database type field from the ProContext rather than the context itself :thinking: .

Reply:
We could add another associated type to the `ProContext` for a ProDatasetDB with a ProDatasetDb bound and then a new accessor method `pro_dataset_db()` which would be called instead of the regular `dataset_db()`. Other than that I don't see how to solve it and if you have an idea please don't leave me guessing ;)

Reply:
Yeah, that is what I thought.

Reply:
b8f303224a732d82a5e1ba65ce2081429843c069